[WIP] Multi chain support

rperez89 opened the pull request.

rperez89 committed:
fix linter

rperez89 committed:
errrors handle

fabriziovigevani commented on src/errors.ts:
```diff
@@ -1,16 +1,21 @@
-      `Required chain: ${getNetworkName(
-        expectedChainId
-      )} (Chain ID: ${expectedChainId}).`
+      `Supported chains: ${supportedChainIds.map((chainId) =>
+        getNetworkName(chainId)
+      )} (Chain ID: ${supportedChainIds}).`
```
Maybe we can group each chain name with their id.

```suggestion
      `Supported chains: ${supportedChainIds.map((chainId) =>
        `(${getNetworkName(chainId)}, ID: ${chainId})`
      ).join(', ')}.`
```

fabriziovigevani commented on src/index.tsx:
```diff
@@ -52,7 +53,7 @@ type WalletContext = {
 
 type UseWalletProviderProps = {
-  chainId: number
+  supportedChains: [number]
```
Let's alphabetize everywhere we have updated it.

fabriziovigevani commented on src/index.tsx:
```diff
@@ -255,6 +256,7 @@ function UseWalletProvider({
     throw new Error('<UseWalletProvider /> has already been declared.')
   }
 
+  const [chainId, setChainId] = useState<number>(1)
```
Should be `-1` right?
```suggestion
  const [chainId, setChainId] = useState<number>(-1)
```

fabriziovigevani commented on src/index.tsx:
```diff
@@ -276,6 +278,7 @@ function UseWalletProvider({
     if (web3ReactContext.active) {
       web3ReactContext.deactivate()
     }
+    setChainId(1)
```
```suggestion
    setChainId(-1)
```

fabriziovigevani commented on src/index.tsx:
```diff
@@ -310,10 +312,14 @@ function UseWalletProvider({
       // Initialize the web3-react connector if it exists.
       const web3ReactConnector = connector?.web3ReactConnector?.({
-        chainId,
+        supportedChains,
```
Looking at the rest of connectors other than `frame`, `provided` and `injected`, they accept a `chainId` prop, in that case we might want to update those connectors to accept the `supportedChains` prop and pass as the chainId the first one as default? 

Was thinking how we can make those connectors support multichain but unless we provide a way for the user to provide which chain they would like to connect to, don't see an alternative.

rperez89 committed:
Connectors Update

We needed to update the connectors that only accepts one chainId to
start receiving the array of supported chainId and default it to use the
first element to keep the library working

rperez89 committed:
pr comments

fabriziovigevani reviewed (approved):
LGTM!
